rpy/robjects/numpy2ri.py

```diff
@@ -53,14 +53,15 @@ def numpy_O_py2rpy(o):
 def numpy2rpy(o):
     """ Augmented conversion function, converting numpy arrays into
     rpy2.rinterface-level R structures. """
-    if not o.dtype.isnative:
+    if (hasattr(o.dtype, 'isnative') and not o.dtype.isnative) or \
+       (hasattr(o.dtype, 'numpy_dtype') and not o.dtype.numpy_dtype.isnative):
```

Comment:
Wait, `o.dtype` is then a `pandas` object then. This converter is for `numpy`.

Reply:
yes this is a `pandas` object, but the pandas code hands it over to the numpy parsing, see above.

Reply:
The change in the PR is leaking pandas-specific rules into a converter that should only cover `numpy`. For example, unless I am mistaken `numpy` arrays do not have `isnative` attributes.

Reply:
I'm open to suggestions on how to do this properly.

Reply:
Move pandas-specific conversion choices to the pandas converter.

Reply:
but isn't the code currently designed to handle a pandas array as a numpy array? Or is the parsing accidentally ending up in the numpy module?

Reply:
i've just pushed changes that hopefully look much better.